src/ci/docker/scripts/fuchsia-test-runner.py

```diff
@@ -676,19 +683,22 @@ def log(msg):
             log("Publishing package to repo...")
 
             # Publish package to repo
-            subprocess.check_call(
-                [
-                    self.tool_path("pm"),
-                    "publish",
-                    "-a",
-                    "-repo",
-                    self.repo_dir(),
-                    "-f",
-                    far_path,
-                ],
-                stdout=log_file,
-                stderr=log_file,
-            )
+            with open(self.pm_lockfile_path(), 'w') as pm_lockfile:
+                fcntl.lockf(pm_lockfile.fileno(), fcntl.LOCK_EX)
+                subprocess.check_call(
+                    [
+                        self.tool_path("pm"),
+                        "publish",
+                        "-a",
+                        "-repo",
+                        self.repo_dir(),
+                        "-f",
+                        far_path,
+                    ],
+                    stdout=log_file,
+                    stderr=log_file,
+                )
+                fcntl.lockf(pm_lockfile.fileno(), fcntl.LOCK_UN)
```

Comment:
This shouldn't be necessary, since the lock should be release as soon as the file is closed. If it was necessary I'd be concerned about what happens when the `check_call` above throws an exception.

Reply:
I agree. I'd prefer to keep it though, since the wording for `fcntl` leaves some wiggle room:

> Leases are associated with an open file description (see
> [open(2)](https://man7.org/linux/man-pages/man2/open.2.html)).  This means that duplicate file descriptors (created
> by, for example, [fork(2)](https://man7.org/linux/man-pages/man2/fork.2.html) or [dup(2)](https://man7.org/linux/man-pages/man2/dup.2.html)) refer to the same lease, and
> this lease may be modified or released using any of these
> descriptors.  Furthermore, the lease is released by either an
> explicit F_UNLCK operation on any of these duplicate file
> descriptors, or when all such file descriptors have been closed.

I don't think any of those exceptions are likely to occur, but explicitly unlocking makes me feel safer / seems generally responsible. When we throw an exception, it's reasonable to expect the process to exit shortly afterward so we won't lose significant performance if we don't unlock in that case.

I'll add a comment explaining that it isn't strictly necessary.